central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -257,3 +252,41 @@ func getScopeKeys(scanClusters []*storage.ComplianceOperatorScanConfigurationV2_
 
 	return clusterScopeKeys
 }
+
+func (ds *datastoreImpl) deleteClusterFromScanConfigWithLock(ctx context.Context, clusterID string, scanConfig *storage.ComplianceOperatorScanConfigurationV2) error {
+	ds.keyedMutex.Lock(scanConfig.GetId())
```

Comment:
nit:  Can you throw some line breaks in here?  Break up logical blocks.  like new line after the defer of the lock.  new line before call to delete by query.

Also some comments around the clusters and filter function would be helpful